packages/components/src/mobile/dark-mode/index.native.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * External dependencies
+ */
+import { eventEmitter, initialMode } from 'react-native-dark-mode';
+import React from 'react';
+
+// Conditional needed to pass UI Tests on CI
+if ( eventEmitter.setMaxListeners ) {
+	eventEmitter.setMaxListeners( 150 );
+}
+
+export function withTheme( WrappedComponent ) {
```

Comment:
Using `theme` for this could be misleading, as WordPress has already a "theme" concept for a different thing. I think we could borrow the term "preferred color scheme" from the web implementation of dark mode and implement this cross platform?

Just like we have a `useReducedMotion` hook, we could also have a `usePreferredColorScheme`. For now, since you can't use hooks in class components, we could still have a `withPreferredColorScheme` HOC for this.

The web implementation seems simple enough:

```jsx
const isDarkMode = useMediaQuery( '(prefers-color-scheme: dark)' );
```

Reply:
Agreed, I much prefer this approach 👍

Reply:
`withPreferredColorScheme` - sounds like a good one. We still might want to refactor the hook as follows:

```js
const isDarkMode = usePreferredColorScheme( 'dark' );
```
and use it inside the HOC. `useMediaQuery` might be not applicable to native mobile, right?